### ISSUE TITLE
Report: JAMA Style is out of date

### ISSUE THREAD

**Issue by alnajar** [ddsjoberg/gtsummary#1590]
I want to contribute to `gtsummary` by updating JAMA theme. I got published recently in one of JAMA Network journals and noticed multiple things that needed to be changed in order to help with the submission. I worked on updating the style as follows:

```
library(tidyverse)
library(gtsummary)

my_theme <-
  list(
    "pkgwide-str:theme_name" = "JAMA/Submission",
    "pkgwide-fn:pvalue_fun" = function(x) style_pvalue(x, digits = 2),
    "pkgwide-fn:prependpvalue_fun" = function(x) style_pvalue(x, digits = 2, prepend_p = TRUE),
    "pkgwide-str:ci.sep" = " \U2013 ",
    "style_number-arg:decimal.mark" = ".",
    "style_number-arg:big.mark" = ",",
    "add_stat_label-arg:location" = "row",
    "tbl_summary-str:continuous_stat" = "{median} ({p25} \U2013 {p75})",
    "tbl_summary-str:categorical_stat" = "{n} ({p})",
    "tbl_summary-fn:addnl-fn-to-run" = function(x) {
      add_stat_label(x)
    },
    "pkgwide-fun:pre_conversion" = function(x) {
      x %>%
      modify_table_styling(
        columns = everything(),
        rows = !is.na(.data$variable),
        missing_symbol = "NA"
      )
    },
    "add_difference-fn:addnl-fn-to-run" = function(x) {
      # merging coef and CI columns, if error, returning x unaltered
      tryCatch(
        {
          new_header_text <-
            paste0(
              x$table_styling$header %>% filter(.data$column == "estimate") %>% pull("label"),
              " **(**",
              x$table_styling$header %>% filter(.data$column == "ci") %>% pull("label"),
              "**)**"
            )
          # adding CI footnote to any existing abbreviation footnote, e.g. for OR, HR, etc.
          estimate_footnote <-
            x$table_styling$footnote_abbrev %>%
            filter(.data$column %in% "estimate") %>%
            filter(dplyr::row_number() == dplyr::n(), !is.na(.data$footnote)) %>%
            dplyr::pull("footnote") %>%
            c("CI = Confidence Interval") %>%
            paste(collapse = ", ")
          x %>%
            # merge estimate and CI into one cell
            modify_table_styling(
              columns = "estimate",
              rows = !!expr(.data$variable %in% !!x$table_body$variable &
                              !is.na(.data$estimate)),
              cols_merge_pattern = "{estimate} ({conf.low} \U2013 {conf.high})"
            ) %>%
            # hide ci column
            modify_column_hide(any_of("ci")) %>%
            # update column header
            modify_header(list(estimate = new_header_text)) %>%
            # add CI abbreviation footnote
            modify_footnote(estimate ~ estimate_footnote, abbreviation = TRUE)
        },
        error = function(e) x
      )
    },
    "tbl_regression-fn:addnl-fn-to-run" = function(x) {
      # merging coef and CI columns, if error, returning x unaltered
      tryCatch(
        {
          new_header_text <-
            paste0(
              x$table_styling$header %>% filter(.data$column == "estimate") %>% pull("label"),
              " **(", style_number(x$inputs$conf.level, scale = 100), "% CI)**"
            )
          # adding CI footnote to any existing abbreviation footnote, e.g. for OR, HR, etc.
          estimate_footnote <-
            x$table_styling$footnote_abbrev %>%
            filter(.data$column %in% "estimate") %>%
            filter(dplyr::row_number() == dplyr::n(), !is.na(.data$footnote)) %>%
            dplyr::pull("footnote") %>%
            c("CI = Confidence Interval") %>%
            paste(collapse = ", ")
          x %>%
            # merge estimate and CI into one cell
            modify_table_styling(
              columns = "estimate",
              rows = !!expr(.data$variable %in% !!x$table_body$variable &
                              !is.na(.data$estimate) &
                              !.data$reference_row %in% TRUE),
              cols_merge_pattern = "{estimate} ({conf.low} \U2013 {conf.high})"
            ) %>%
            # Reference
            modify_table_styling(
              columns = estimate,
              rows = reference_row %in% TRUE,
              missing_symbol = "1 [Reference]"
            ) %>% 
            # hide ci column
            modify_column_hide(any_of("ci")) %>%
            # update column header
            modify_header(list(estimate = new_header_text)) %>%
            # add CI abbreviation footnote
            modify_footnote(estimate ~ estimate_footnote, abbreviation = TRUE) 
        },
        error = function(e) x
      )
    }
  )

check_gtsummary_theme(my_theme)
set_gtsummary_theme(my_theme)
theme_gtsummary_compact()
# reset_gtsummary_theme()
```

Examples

```
trial %>%
  select(age, response, grade) %>%
  tbl_summary(
    statistic = all_continuous() ~ "{mean} ({sd})",
    type = all_dichotomous() ~ "categorical"
  )
```
![image](https://github.com/ddsjoberg/gtsummary/assets/7356035/ba5d3cee-d10f-462f-8ecc-31842be5247d)

```
glm(response ~ age + stage, trial, family = binomial)  %>%
tbl_regression ( exponentiate = TRUE)
```
![image](https://github.com/ddsjoberg/gtsummary/assets/7356035/f71975e9-9908-4613-9702-3f49a15cf7d2)

**Comment by alnajar** [comment]
Additional points:

I need to change [footnote marks as subscript instead of superscript](https://stackoverflow.com/questions/76956929/gtsummary-footnote-marks-as-subscript-instead-of-superscript)

I made a `[stackoverflow](https://stackoverflow.com/questions/77868390/how-to-change-gtsummary-themes-footnote-marks-as-subscript-instead-of-superscri)` question and would appreciate any help
https://stackoverflow.com/questions/77868390/how-to-change-gtsummary-themes-footnote-marks-as-subscript-instead-of-superscri

**Comment by ddsjoberg** [comment]
Thanks for the updates @alnajar !

FYI, the package is undergoing to larger internal changes at the moment. So it may be a few months until this is reviewed.

In the meantime, if you added more detail to your examples that would be helpful:
1. What have you added or modified?
2. Please include a link to the JAMA guidelines indicating the need for the change

Regarding your SO post: If you provide the code in gt to change a superscript to a subscript, I can look into the details of adding it to a theme.

Thanks,

**Comment by alnajar** [comment]
I changed the following: 

1. confidence interval range from "to" to "-", as per their instructions:
> Table (and throughout): Express 95% CIs as ranges: X.XX-Y.YY. Combine HR and 95% CI columns as HR (95% CI). 

2. all empty cells will become "NA".
3. all references changed from "_" to "1 [Reference]",  as per their instructions:
> Table: All data cells should include “NA” if intentionally left blank or, in the case of reference group HR (95% CI), should included “1 [Reference]”. 

Hopefully the community and I, with your assistance, maybe we solve my above question before next release in order to change the footnotes into letters as well as prevent `bold_p` from working and have `bold_label` as default. Per their instructions:
> Table (and throughout): Statistical significance cannot be indicated via asterisks, formatting, or footnotes. Please remove all bolding from the table; the P values themselves are sufficient. Express P values to 3 decimals if less than .01 or to 2 decimals if greater than or equal to .01. Please ensure that the level of statistical significance is stated within Methods. 

> [Missing data and blank space in the table field (ie, an empty cell) may create ambiguity and should be avoided; use abbreviations such as NA for not applicable or not available. Each piece of data needs to be contained in its own cell. ] (https://jamanetwork.com/journals/jama/pages/instructions-for-authors#SecTables)

> [Use superscript letters (a, b, c) to mark each footnote and be sure each footnote in the table has a corresponding note (and vice versa). ] (https://jamanetwork.com/journals/jama/pages/instructions-for-authors#SecTables)

> [Headings within tables should be set in 10 point and bold. Table footnotes should be set in 8 point and single-spaced] (https://jamanetwork.com/journals/jama/pages/instructions-for-authors#SecTables)

**Comment by ddsjoberg** [comment]
Hi @alnajar !

Thanks for the details!

> 1. confidence interval range from "to" to "-", as per their instructions:

I didn't see anything in the author instructions about using a hyphen instead of "to" as the separator in a table. They do give example of using "to" in the body of a manuscript, however. If I recall correctly, I added "to" because I reviewed a few papers in the current edition (at the time of implementing), and they all used "to". I took a look at the most recent issue and an article they highlighted on their homepage, and it has a mix of "to" and hyphen for CI separators 🤷🏼 I am not against a change per se, but I don't like the idea of changing when the don't have a clear and consistent policy? What if the next reviewer requests "to" instead of hyphen?

![image](https://github.com/ddsjoberg/gtsummary/assets/26774684/b0dc9e3e-2c09-41a5-9af2-27b23129d2bc)

> 2. all empty cells will become "NA".

I don't think they mean alllll empty cells. Even the examples on the author instruction page have blanks, where it's clear they are meant to remain blank.
![image](https://github.com/ddsjoberg/gtsummary/assets/26774684/f28cdebf-3bf5-4157-873d-d0579fd983d3)

> 3. all references changed from "_" to "1 [Reference]", as per their instructions:

Where did you get the request for a change? I didn't see it in the Author Instruction page.

> 4. Hopefully the community and I, with your assistance, maybe we solve my above question before next release in order to change the footnotes into letters as well as prevent bold_p from working and have bold_label as default. 

I don't think we need to prevent `bold_p()` calls: we just don't make it the default. We can add `bold_labels()` as a default call when the JAMA theme is applied, however. Regarding the footnotes, this is something we'll need to investigate further.

Thanks again!!!
